packages/flutter/test/material/search_bar_theme_test.dart

```diff
@@ -0,0 +1,318 @@
+  });
+
+  group('[Theme, SearchBarTheme, SearchBar properties overrides]', () {
+    const double elevationValue = 5.0;
+    const Color backgroundColorValue = Color(0xff000001);
+    const Color shadowColorValue = Color(0xff000001);
+    const Color surfaceTintColorValue = Color(0xff000001);
+    const Color overlayColorValue = Color(0xff000001);
+    const BorderSide sideValue = BorderSide(
+        color: Color(0xff000004), width: 2.0);
+    const OutlinedBorder shapeValue = RoundedRectangleBorder(
+        side: sideValue, borderRadius: BorderRadius.all(Radius.circular(2.0)));
+    const EdgeInsets paddingValue = EdgeInsets.symmetric(horizontal: 16.0);
+    const TextStyle textStyleValue = TextStyle(
+        color: Color(0xff000005), fontSize: 20.0);
+    const TextStyle hintStyleValue = TextStyle(
+        color: Color(0xff000006), fontSize: 18.0);
+
+    const MaterialStateProperty<double?> elevation = MaterialStatePropertyAll<
+        double>(elevationValue);
+    const MaterialStateProperty<
+        Color?> backgroundColor = MaterialStatePropertyAll<Color>(
+        backgroundColorValue);
+    const MaterialStateProperty<Color?> shadowColor = MaterialStatePropertyAll<
+        Color>(shadowColorValue);
+    const MaterialStateProperty<
+        Color?> surfaceTintColor = MaterialStatePropertyAll<Color>(
+        surfaceTintColorValue);
+    const MaterialStateProperty<Color?> overlayColor = MaterialStatePropertyAll<
+        Color>(overlayColorValue);
+    const MaterialStateProperty<BorderSide?> side = MaterialStatePropertyAll<
+        BorderSide>(sideValue);
+    const MaterialStateProperty<
+        OutlinedBorder?> shape = MaterialStatePropertyAll<OutlinedBorder>(
+        shapeValue);
+    const MaterialStateProperty<
+        EdgeInsetsGeometry?> padding = MaterialStatePropertyAll<EdgeInsets>(
+        paddingValue);
+    const MaterialStateProperty<
+        TextStyle?> textStyle = MaterialStatePropertyAll<TextStyle>(
+        textStyleValue);
+    const MaterialStateProperty<
+        TextStyle?> hintStyle = MaterialStatePropertyAll<TextStyle>(
+        hintStyleValue);
+    const BoxConstraints constraints = BoxConstraints(
+        minWidth: 250.0, maxWidth: 300.0, minHeight: 80.0);
+
+    const SearchBarThemeData searchBarTheme = SearchBarThemeData(
+      elevation: elevation,
+      backgroundColor: backgroundColor,
+      shadowColor: shadowColor,
+      surfaceTintColor: surfaceTintColor,
+      overlayColor: overlayColor,
+      side: side,
+      shape: shape,
+      padding: padding,
+      textStyle: textStyle,
+      hintStyle: hintStyle,
+      constraints: constraints,
+    );
+
+    Widget buildFrame(
+        { bool useSearchBarProperties = false, SearchBarThemeData? searchBarThemeData, SearchBarThemeData? overallTheme }) {
+      final Widget child = Builder(
+        builder: (BuildContext context) {
+          return !useSearchBarProperties
```

Comment:
Large `: ?` expressions like this are often easier to read with a short-circuit return:
```dart
if (!useSearchBarProperties) {
  return SearchBar(
    ...
    );
}
return SearchBar(
  ...
);
```